TABTestKit/Classes/Contexts/NavigationContext.swift

```diff
@@ -53,11 +53,44 @@ public extension NavigationContext {
 	///
 	/// - Parameter dismissableThings: One or more Dismissable things. Typically, this would be a Screen that conforms to Dismissable.
 	func dismiss(_ dismissableThings: Dismissable...) {
-		guard !dismissableThings.isEmpty else { XCTFatalFail("You must provide at least one Dismissable thing to dismizs!") }
+		guard !dismissableThings.isEmpty else { XCTFatalFail("You must provide at least one Dismissable thing to dismiss!") }
 		dismissableThings.forEach {
 			$0.await()
 			$0.dismiss()
 		}
 	}
+
+  /// Opens the specified URL from the test runner app.
+  /// This means you can use this function for testing deep linking, as well as opening web urls in Safari.
+  /// This function will first show the Home Screen if necessary, launch the test runner (after finding the correct
+  /// page with the runner) and then open the URL.
+  ///
+  /// It is your responsibility/choice if you want to leave your app in the current state before the test runner
+  /// opens the URL, or whether you want to completely terminate your app before hand.
+  ///
+  /// This leaves you free to test deep linking behaviour from the current state of the app.
+  /// or a fresh launch of the app.
+  ///
+  /// Since you can pass any URL to this function, you could pass an app-specific url like `myapp://deep-link`, or
+  /// a regular HTTP link like `https://apple.com`.
+  ///
+  /// HTTP links are opened in Safari by default, unless you have an app installed that can intercept them using
+  /// Universal Links.
+  ///
+  /// - Parameter url: The URL to open.
+  func open(_ url: URL) {
+   springboard.activate()
+    if !Icon.testRunner.underlyingXCUIElement.isHittable {
+      XCUIDevice.shared.press(.home) // Ensure we're on the first page of the home screen
+      sleep(1)
+      while !Icon.testRunner.underlyingXCUIElement.isHittable {
+        springboard.swipeLeft()
+        sleep(1)
+      }
+    }
+    Icon.testRunner.tap()
+    sleep(1)
+    UIApplication.shared.open(url)
```

Comment:
Is it worth dealing with the open permission Alert you will have to approve the first time and may have to approve subsequent times as part of this function?

I can't see a use case where a user wouldn't want to give permission.

Reply:
Updated